pkg/git/options.go

```diff
@@ -73,6 +73,11 @@ type AuthOptions struct {
 	Identity   []byte
 	KnownHosts []byte
 	CAFile     []byte
+	// TransportAuthID is a unique identifier for this set of authentication
+	// options. It's used by managed libgit2 transports to uniquely identify
+	// which credentials to use for a particular git operation, and avoid misuse
+	// of credentials in a multi tenant environment.
```

Comment:
Could expand on the required format here - must be based on protocols registered by managed transport.